FOLIO Wrap-up - Remove Item from New Bookshelf Workflow (#345)
TAMULib/fw-registry: README.md

- Be sure to check and update the tenant header in all the curl requests documented below.

## Variable substitution

> The workflow JSON files are templates initially processed by `fw-cli` using node handlebars template engine followed by Java delegate expression value variable substitution done with [FreeMarkerTemplateUtils](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/ui/freemarker/FreeMarkerTemplateUtils.html) in `mod-camunda` Java delegates. These JSON files are also a `fw-cli` representation of `mod-workflow` domain model which is heavily influenced on [BPMN](https://www.bpmn.org/) and [Camunda](https://camunda.com/).

Author: ```suggestion
The workflow JSON files are templates initially processed by `fw-cli` using node handlebars template engine followed by Java delegate expression value variable substitution done with [FreeMarkerTemplateUtils](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/ui/freemarker/FreeMarkerTemplateUtils.html) in `mod-camunda` Java delegates and [JUEL](https://juel.sourceforge.net/) used by the Camunda BPMN engine. These JSON files are also a `fw-cli` representation of `mod-workflow` domain model which is heavily influenced on [BPMN](https://www.bpmn.org/) and [Camunda](https://camunda.com/).
```

***These hold credentials/secrets that conf node package used by fw-cli to store configurations loaded into a user home app data directory.***

1. `{{{}}}` and `{{}}`
Syntax for [handlebars](https://handlebarsjs.com/) template processing in `fw-cli` at workflow build using `fw build`.
1. `${}`

> Syntax for [freemarker](https://freemarker.apache.org/) and [JUEL](https://juel.sourceforge.net/) template processing in `mod-camunda` at workflow activate. Also used by [camunda expressions](https://docs.camunda.org/manual/7.20/user-guide/process-engine/expression-language/) evaluation which affords calling methods on the object at runtime of the workflow.

Author: ```suggestion
Syntax for [freemarker](https://freemarker.apache.org/) and [JUEL](https://juel.sourceforge.net/) template processing in `mod-camunda` at workflow run. Also used by [camunda expressions](https://docs.camunda.org/manual/7.20/user-guide/process-engine/expression-language/) evaluation which affords calling methods on the object at runtime of the workflow.
```

  --header 'X-Okapi-Tenant: diku' \
  --form 'logLevel="INFO"' \
  --form 'emailFrom="folio@k1000.library.tamu.edu"' \
  --form 'emailTo="workflows@library.tamu.edu"' \
  --form 'file=@"itemBarcodes.csv"' \
  --form 'path="/mnt/workflows/diku/create-notes/"' \
  --form 'itemNoteTypeId="d5684236-e4ab-4a64-97b3-2aa7a595cfc4"' \
  --form 'noteText="This is a note text message."' \
  --form 'staffOnly=false' \
  --form 'username="***"' \
  --form 'password="***"'
```

## remove-books-from-nbs

> Remove items from new bookshelf location for provided CSV of call numbers.

Author: ```suggestion
For the uploaded CSV of call numbers, remove items that have been on the new bookshelf location for more than 30 days.
```
